bpo-31109: Convert zipimport to argument clinic

jarondl opened the pull request:
Convert the zipimport module to argument clinic. Mostly
straight-forward, had to rename a few argument names.

<!-- issue-number: bpo-31109 -->
https://bugs.python.org/issue31109
<!-- /issue-number -->

jarondl committed:
bpo-31109: Convert zipimport to argument clinic

Convert the zipimport module to argument clinic. Mostly
straight-forward, had to rename a few argument names.

serhiy-storchaka commented on Modules/zipimport.c:
```diff
@@ -54,46 +54,72 @@ static PyObject *get_data(PyObject *archive, PyObject *toc_entry);
+/*[clinic input]
+zipimport.zipimporter.__init__
+
+    archivepath: object(converter="PyUnicode_FSDecoder")
```
Why not use just the name "path"?

jarondl replied:
Do you mean `path:` or `archivepath as path:`? 
Sure I can do the latter, but what's the benefit? It was more straight forward to rename the variable to match the signature. The meaning of `archivepath` is clearer

brettcannon replied:
The benefit of not renaming is the patch is much easier to review that way. As it stands it's much harder for us to check the accuracy of the changes due to so many lines being flagged as changed compared to just reviewing the Argument Clinic changes which is what this PR is specifically targeting.

jarondl replied:
Done.

serhiy-storchaka commented on Modules/zipimport.c:
```diff
@@ -379,21 +405,29 @@ find_loader(ZipImporter *self, PyObject *fullname, PyObject **namespace_portion)
+/*[clinic input]
+zipimport.zipimporter.find_module
+
+    fullname: unicode
```
The arguments of this method (and many other methods) were positional-only. Converting to Argument Clinic shouldn't change the behavior.

jarondl replied:
Actually *all* the methods here were positional-only. But this change is backwards compatible and logical. See for example http://bugs.python.org/issue20294#msg208444
Why should we force positional-only? Your patch in bpo-25711 to convert `zipimport` to Python would also have the same effect.

jarondl replied:
Do you prefer to keep Argument Clinic conversion separate from behavioral changes? Then we can add trailing `/`s to the argument lists and remove them in a separate PR.

brettcannon replied:
Yeah, we prefer to not change semantics while doing an Argument Clinic conversion (we have run into issues in the past when doing both at the same time).

jarondl replied:
OK, done, now they are positional only. Thank you both for reviewing.

jarondl committed:
bpo-31109: zipimport argument clinic: positional only

Preserve positional only arguments in zipimport after argument clinic
conversion.

jarondl committed:
bpo-31109: zipimport argument clinic: revert arg names

Revert the argument names archivepath and pathname to their
original names in the code (path, and path).

brettcannon commented on Modules/zipimport.c:
```diff
@@ -54,28 +54,54 @@ static PyObject *get_data(PyObject *archive, PyObject *toc_entry);
+'/tmp/myimport.zip/mydirectory', if mydirectory is a valid directory inside
+the archive.
+
+'ZipImportError is raised if 'archivepath' doesn't point to a valid Zip
```
Looks like there is a stray `'` at the start of this sentence.

brettcannon commented on Modules/zipimport.c:
```diff
@@ -54,28 +54,54 @@ static PyObject *get_data(PyObject *archive, PyObject *toc_entry);
+'ZipImportError is raised if 'archivepath' doesn't point to a valid Zip
+archive.
+
+The 'archive' attribute of zipimporter objects contains the name of the
```
"zipimporter objects" -> "the zipimporter object".

brettcannon commented on Modules/zipimport.c:
```diff
@@ -416,23 +451,32 @@ zipimporter_find_module(PyObject *obj, PyObject *args)
+
+Search for a module specified by 'fullname'.
+
+'fullname' must be the
```
Premature line break.

brettcannon reviewed (changes requested):
Three really minor formatting tweaks necessary and then I think this will be ready to go!

jarondl committed:
bpo-31109: zipimporter docstrings minor formatting fixes

brettcannon reviewed (approved):
@serhiy-storchaka did you have any other feedback or can I merge this?

jarondl commented:
@serhiy-storchaka and @brettcannon , is there anything else I can do to have this merged?
In any case I want to say thank you both for the reviews.

brettcannon commented:
Since it looks like @serhiy-storchaka has nothing to add would you mind adding a news entry? See https://devguide.python.org/committing/#what-s-new-and-news-entries on how to do that. Once that's file in I can touch it up as necessary and then merge this!

brettcannon reviewed (changes requested):
Please add a news entry.

jarondl committed:
bpo-31109: add NEWS entry

jarondl commented:
Blimey, I didn't expect the Spanish Inquisition!

brettcannon committed:
Very minor news entry tweak

brettcannon reviewed (approved):
Once Travis goes green I'll merge!

brettcannon commented:
Thanks for the work, @jarondl !
